source/ui/dataman/AMSpectrumAndPeriodicTableView.cpp

```diff
@@ -0,0 +1,388 @@
+#include "AMSpectrumAndPeriodicTableView.h"
+
+#include "dataman/AMScan.h"
+#include "ui/dataman/AMColoredTextToolButton.h"
+#include "ui/util/AMPeriodicTableDialog.h"
+
+#include "MPlot/MPlotTools.h"
+
+#include <QStringBuilder>
+
+ AMSpectrumAndPeriodicTableView::~AMSpectrumAndPeriodicTableView(){}
+AMSpectrumAndPeriodicTableView::AMSpectrumAndPeriodicTableView(QWidget *parent)
+	: QWidget(parent)
+{
+	emissionLineValidator_ = new AMNameAndRangeValidator(this);
+	pileUpPeakValidator_ = new AMNameAndRangeValidator(this);
+	combinationPileUpPeakValidator_ = new AMNameAndRangeValidator(this);
+
+	periodicTable_ = new AMSelectablePeriodicTable(this);
+	periodicTable_->buildPeriodicTable();
+
+	connect(periodicTable_, SIGNAL(elementSelected(AMElement*)), this, SLOT(onElementSelected(AMElement*)));
+	connect(periodicTable_, SIGNAL(elementDeselected(AMElement*)), this, SLOT(onElementDeselected(AMElement*)));
+
+	periodicTableView_ = new AMSelectablePeriodicTableView(periodicTable_);
+	periodicTableView_->buildPeriodicTableView();
+	connect(periodicTableView_, SIGNAL(elementSelected(AMElement*)), this, SLOT(onElementClicked(AMElement*)));
+
+	rowAbovePeriodicTableLayout_ = new QHBoxLayout;
+}
+
+void AMSpectrumAndPeriodicTableView::setupPlot()
+{
+	plot_ = new MPlot;
+	plotView_ = new MPlotWidget(this);
+	plotView_->setPlot(plot_);
+
+	plotView_->plot()->plotArea()->setBrush(QBrush(Qt::white));
+	plotView_->plot()->axisBottom()->setAxisNameFont(QFont("Helvetica", 6));
+	plotView_->plot()->axisBottom()->setTickLabelFont(QFont("Helvetica", 6));
+	plotView_->plot()->axisBottom()->showAxisName(true);
+	plotView_->plot()->axisLeft()->showAxisName(false);
+
+	plotView_->plot()->addTool(new MPlotDragZoomerTool());
+	plotView_->plot()->addTool(new MPlotWheelZoomerTool());
+
+	plotView_->setSizePolicy(QSizePolicy::MinimumExpanding, QSizePolicy::MinimumExpanding);
+
+	plot_->axisScaleLeft()->setDataRangeConstraint(MPlotAxisRange(0, MPLOT_POS_INFINITY));
+
+	currentElement_ = periodicTable_->elementBySymbol("Fe");
+	combinationElement_ = periodicTable_->elementBySymbol("Ca");
+
+	QPushButton *removeAllEmissionLinesButton = new QPushButton(QIcon(":/trashcan.png"), "Clear Emission Lines");
+	removeAllEmissionLinesButton->setMaximumHeight(25);
+	rowAbovePeriodicTableLayout_->addWidget(removeAllEmissionLinesButton);
+	connect(removeAllEmissionLinesButton, SIGNAL(clicked()), this, SLOT(removeAllEmissionLineMarkers()));
+}
+
+void AMSpectrumAndPeriodicTableView::buildEnergyRangeSpinBoxView()
+{
+	minimum_ = new QDoubleSpinBox;
+	minimum_->setSuffix(" eV");
+	minimum_->setDecimals(0);
+	minimum_->setRange(0, 1000000);
+	connect(minimum_, SIGNAL(editingFinished()), this, SLOT(onMinimumEnergyChanged()));
+
+	maximum_ = new QDoubleSpinBox;
```

Comment:
Could these be renamed to
maximumEnergySpinBox and minimumEnergySpinBox? Just makes it a little clearer that they are ui elements and not values.